### ISSUE TITLE
Wrong document for dynamic_rnn

### ISSUE THREAD

**Issue by ruanchong** [tensorflow/tensorflow#27981]
<em>Please make sure that this is a documentation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:doc_template</em>

**System information**
- TensorFlow version: TF r1.13 (and many other versions)
- Doc Link: [tf.nn.dynamic_rnn](https://www.tensorflow.org/api_docs/python/tf/nn/dynamic_rnn)

**Describe the documentation issue**

It describes the argument `sequence_length` as "it's more for performance than correctness", but this is wrong:

- This argument enables this API to extract the last **VALID** state of RNN instead of a **PADDED** time step, **so it IS for correctness.**
- This argument CANNOT reduce computation, because even with this argument, computation of new states of RNN cell still occurs. The only difference is that `tf.nn.dynamic_rnn` chooses to copy through old states instead of keeping the new states, and this leads to even larger computation. **So it is NOT for performance.**

The initial document is correct, see [this commit](https://github.com/tensorflow/tensorflow/commit/855d3b56014780a90143b3e0c0865334b188c2df).

> it's more for correctness than performance"

I don't really understand why you replaced the correct document with a wrong one.

**We welcome contributions by users. Will you be able to update submit a PR (use the [doc style guide](https://www.tensorflow.org/community/documentation)) to fix the doc Issue?**

**Comment by muddham** [comment]
@ruanchong We will take a look at it . Thanks!

**Comment by zakizhou** [comment]
@ruanchong is correct, and `sequence_length` indeed is for correctness not for performance, so is `mask` in `tf.keras.layers.RNN.__call__`
